File: problems/Monotonic Array/isMonotonic.cs
public class Solution {
    public bool IsMonotonic(int[] A) {
        if (1 == A.Length) {
            return true;
        }

        var diff = A[0] - A[1];

        for (var i = 1; A.Length > i; ++i) {
            var currDiff = A[i - 1] - A[i];

            if (0 == currDiff) {
                continue;
            } else if (0 == diff) {
                diff = currDiff;
            } else if (Math.Sign(diff) != Math.Sign(currDiff)) {
                return false;
            }
        }

        return true;
    }
}

File: problems/Monotonic Array/isMonotonic.java
class Solution {
    public boolean isMonotonic(int[] A) {
        int comp = 0;

        for (int idx = 1; A.length > idx; ++idx) {
            if (0 == comp) {
                comp = A[idx - 1] - A[idx];
            } else if (0 < comp && A[idx - 1] < A[idx]) {
                return false;
            } else if (0 > comp && A[idx - 1] > A[idx]) {
                return false;
            }
        }

        return true;
    }
}
